Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -837,26 +842,40 @@ void CGameListCtrl::OnRightClick(wxMouseEvent& event)
 		if (selected_iso)
 		{
 			wxMenu popupMenu;
-			popupMenu.Append(IDM_PROPERTIES, _("&Properties"));
-			popupMenu.Append(IDM_GAME_WIKI, _("&Wiki"));
-			popupMenu.AppendSeparator();
 
-			if (selected_iso->GetPlatform() != DiscIO::IVolume::GAMECUBE_DISC)
+			if (!selected_iso->IsElfOrDol())
+			{
+				popupMenu.Append(IDM_PROPERTIES, _("&Properties"));
+				popupMenu.Append(IDM_GAME_WIKI, _("&Wiki"));
+				popupMenu.AppendSeparator();
+			}
+			if (selected_iso->GetPlatform() != DiscIO::IVolume::GAMECUBE_DISC && !selected_iso->IsElfOrDol())
 			{
 				popupMenu.Append(IDM_OPEN_SAVE_FOLDER, _("Open Wii &save folder"));
 				popupMenu.Append(IDM_EXPORT_SAVE, _("Export Wii save (Experimental)"));
 			}
 			popupMenu.Append(IDM_OPEN_CONTAINING_FOLDER, _("Open &containing folder"));
-			popupMenu.AppendCheckItem(IDM_SET_DEFAULT_ISO, _("Set as &default ISO"));
+
+			if (!selected_iso->IsElfOrDol())
+				popupMenu.AppendCheckItem(IDM_SET_DEFAULT_ISO, _("Set as &default ISO"));
 
 			// First we have to decide a starting value when we append it
 			if (selected_iso->GetFileName() == SConfig::GetInstance().m_strDefaultISO)
 				popupMenu.FindItem(IDM_SET_DEFAULT_ISO)->Check();
 
-			popupMenu.AppendSeparator();
-			popupMenu.Append(IDM_DELETE_ISO, _("&Delete ISO..."));
+			if (!selected_iso->IsElfOrDol())
+			{
+				popupMenu.AppendSeparator();
+				popupMenu.Append(IDM_DELETE_ISO, _("&Delete ISO..."));
+			}
+			else
+			{
+				popupMenu.AppendSeparator();
+				popupMenu.Append(IDM_DELETE_ISO, _("&Delete File..."));
+			}
+
 
-			if (selected_iso->GetPlatform() != DiscIO::IVolume::WII_WAD)
+			if (selected_iso->SupportsCompression())
```

Comment:
I know someone told you to add the `SupportsCompression function, but it makes the ifs and elses here rather unclear :/
That the final else means "only WADs" is not at all obvious unless you know that `SupportsCompression` means Wii and GC discs and that those combined with DOLs and ELFs are the only existing platforms.